Bottom: a382695e2987cbd0d5a442f0e357413683b8c1df
Top:    0c97026b5f690fd1e2693e9f9c63549fc3e00f00
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-16 15:04:24 -0700

Refresh of waitall-buggy

---

diff --git a/src/rsocket.c b/src/rsocket.c
index 2edd44e..4c4821b 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1171,7 +1171,7 @@ ssize_t rrecv(int socket, void *buf, size_t len, int flags)
 			rs->rbuf_bytes_avail += rsize;
 		}
 
-	} while ((flags & MSG_WAITALL) && (rs->state & rs_connect_rd) && left);
+	} while (left && (flags & MSG_WAITALL) && (rs->state & rs_connect_rd));
 
 	fastlock_release(&rs->rlock);
 	return ret ? ret : len - left;
